### DISCUSSION TITLE
Change logging backend from swift-log to OSLog

### DISCUSSION THREAD

**Discussion by milanvarady** [TheAcharya/MarkersExtractor#85] [Ideas]
@orchetect I have started to implement log collection in MarkerData. What I mean by that is this users will be able to export logs to a `.txt` file at any given time. Now I see that `MarkersExtractor.init()` has a `logger` parameter. The problem is that it uses the `Logging` module and not the older `OSLog`. The reason `OSLog` would be beneficial is that it has an `OSLogStore` object which has a `.getEntries()` method which allows me to get and filter log entries very easily.

My proposal would be to change MarkerExtractor's logging backend to `OSLog`, I think the interface is the same except for initialization. So we would have to change `Logger(label: "\(Object.self)")` to something like `Logger(subsystem: "co.theacharya.MarkersExtractor", category: "\(Object.self)")` everywhere.

**Comment by orchetect** [comment]
Have you tried injecting your own Logger instance to the initializer?

**Comment by milanvarady** [comment]
Yes I have. But the `Logger` from the `Logging` module is not the same as the one from the `OSLog` module. Same name different modules.

I get this error: `Cannot convert value of type 'os.Logger' to expected argument type 'Logging.Logger'`

**Reply by orchetect** [reply]
Use the `swift-log` package and import it where you're creating the `Logging.Logger` instance.

**Reply by milanvarady** [reply]
I know I can do that, but I would like to export the logs out into a file. The OSLog system has `OSLogStore.getEntries()` which allows me to export log entries. If you know of a way the same could be done with `Logger.Logger` then we don't need to change anything.

**Reply by latenitefilms** [reply]
Can you just do this?

`freopen(logPath.cString(using: .ascii), "a+", stderr)`

**Comment by orchetect** [comment]
There is a `swift-log` wrapper that uses OSLog. You can import this and then pass a custom Logger instance in.

https://github.com/chrisaljoudi/swift-log-oslog

**Reply by milanvarady** [reply]
I'll check this out.
